File: ProjetSopraFinal/src/main/java/model/Module.java
package model;

import java.util.Date;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;
import javax.persistence.Version;

@Entity
public class Module {
	
	private Integer id;
	private Salle salle;
	private Formateur formateur;
	private Matiere matiere;
	private VideoProjecteur videoProjecteur;
	private Date dateDebut;
	private Formation formation;
	
	private int version;
	
	public Module() {

	}

	@Id
	@GeneratedValue(strategy = GenerationType.SEQUENCE) // a completer
	public Integer getId() {
		return id;
	}



	public void setId(Integer id) {
		this.id = id;
	}



	public Salle getSalle() {
		return salle;
	}


	public void setSalle(Salle salle) {
		this.salle = salle;
	}


	public Formateur getFormateur() {
		return formateur;
	}


	public void setFormateur(Formateur formateur) {
		this.formateur = formateur;
	}

	@ManyToOne
	@JoinColumn (name = "matiere")
	public Matiere getMatiere() {
		return matiere;
	}


	public void setMatiere(Matiere matiere) {
		this.matiere = matiere;
	}

	@OneToOne
	public VideoProjecteur getVideoProjecteur() {
		return videoProjecteur;
	}


	public void setVideoProjecteur(VideoProjecteur videoProjecteur) {
		this.videoProjecteur = videoProjecteur;
	}

	@Temporal (TemporalType.DATE)
	public Date getDateDebut() {
		return dateDebut;
	}


	public void setDateDebut(Date dateDebut) {
		this.dateDebut = dateDebut;
	}
	
	
	@ManyToOne
	@JoinColumn (name = "formation")
	public Formation getFormation() {
		return formation;
	}

	public void setFormation(Formation formation) {
		this.formation = formation;
	}

	@Version
	public int getVersion() {
		return version;
	}

	public void setVersion(int version) {
		this.version = version;
	}

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + ((id == null) ? 0 : id.hashCode());
		return result;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Module other = (Module) obj;
		if (id == null) {
			if (other.id != null)
				return false;
		} else if (!id.equals(other.id))
			return false;
		return true;
	}
	
	

}

File: AngularSopraFinal/src/app/model/module.ts
import {Salle} from './salle';
import {Formateur} from './formateur';
import {Matiere} from './matiere';
import {VideoProjecteur} from './videoProjecteur';
import {Formation} from './formation';


export class Module {

  constructor(private _id?, private _salle?, private _formateur?, private _matiere?, private _videoProjecteur?,
    private _dateDebut?, private _formation?) {
  }

  public get id(): number {
    return this._id
  }
  public set id(_id: number) {
    this._id = _id;
  }

  public get salle(): Salle {
    return this._salle
  }
  public set salle(_salle: Salle) {
    this._salle = _salle;
  }

  public get formateur(): Formateur {
    return this._formateur
  }
  public set formateur(_formateur: Formateur) {
    this._formateur = _formateur;
  }

  public get matiere(): Matiere {
    return this._matiere
  }
  public set matiere(_matiere: Matiere) {
    this._matiere = _matiere;
  }

  public get videoProjecteur(): VideoProjecteur {
    return this._videoProjecteur
  }
  public set videoProjecteur(_videoProjecteur: VideoProjecteur) {
    this._videoProjecteur = _videoProjecteur;
  }

  public get dateDebut(): Date {
    return this._dateDebut
  }
  public set dateDebut(_dateDebut: Date) {
    this._dateDebut = _dateDebut;
  }

  public get formation(): Formation {
    return this._formation
  }
  public set formation(_formation: Formation) {
    this._formation = _formation;
  }
}